CRM/Activity/BAO/Activity.php

```diff
@@ -2193,30 +2185,25 @@ public static function exportableFields($name = 'Activity') {
           'title' => ts('Activity Type'),
           'type' => CRM_Utils_Type::T_STRING,
         ],
-        'case_activity_status' => [
-          'title' => ts('Activity Status'),
-          'type' => CRM_Utils_Type::T_STRING,
-        ],
-        'case_activity_duration' => [
-          'title' => ts('Activity Duration'),
-          'type' => CRM_Utils_Type::T_INT,
-        ],
         'case_activity_medium_id' => [
           'title' => ts('Activity Medium'),
           'type' => CRM_Utils_Type::T_INT,
         ],
-        'case_activity_details' => [
-          'title' => ts('Activity Details'),
-          'type' => CRM_Utils_Type::T_TEXT,
-        ],
         'case_activity_is_auto' => [
           'title' => ts('Activity Auto-generated?'),
           'type' => CRM_Utils_Type::T_BOOLEAN,
         ],
       ];
+      $caseStandardFields = ['activity_subject', 'source_contact_id', 'activity_status', 'activity_duration', 'activity_details'];
+      foreach ($caseStandardFields as $key) {
+        $caseActivityFields['case_' . $key] = $fields[$key];
+      }
+      // This is a hack to get past an inconsistency.
+      // Activity BAO_Query object adds source_contact as a name field but Case BAO_Query squishes it into the id field.
+      $caseActivityFields['source_contact_id']['type'] = CRM_Utils_Type::T_STRING;
```

Comment:
I should have stated there's two parts of the line that need updating. The whole line needs to be `$caseActivityFields['case_source_contact_id']['type'] = CRM_Utils_Type::T_STRING;`

Reply:
doh! Serves me for trying to fix it while really working on something else

Reply:
No worries.